scalableminds/webknossos
# Fix cursor jumping bug in tree and comment renaming

#7136 removed a workaround for cursor jumping, because I thought that react v16 would have fixed this. During testing, I assumed that all inputs would suffer from this bug, but it turns out that only controlled inputs that store their state in the store (and not in a local state) are affected. I think, this is only true for the tree name and comment inputs which is likely why I missed this back then.
I re-introduced the bug fix and removed the `isFocused` handling, since that assumed that the input's content should not update when the input is changed from the outside while it's focussed. However, the dataset position view supports changing the input via the arrow-keys now which is not compatible with the isFocused check. I think the change should not cause any issues :tm: 

### URL of deployed dev instance (used for testing):
- https://___.webknossos.xyz

### Steps to test:
- create a new tree and try to rename the tree
- create a new node and try to add a comment for it and rename it
- in both cases, do some typical text editing (e.g., also insert something in the middle of the string). the cursor should behave normally (no jumping)
- focus the position input in the navbar and use the arrow keys to increment/decrement the coordinates -> should also work as expected

### Issues:
- fixes #7369

------
(Please delete unneeded items, merge only when none are left open)
